src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/UpdateOperationPage.java

```diff
@@ -5,28 +5,36 @@
 import io.stargate.sgv2.jsonapi.api.model.command.CommandStatus;
 import io.stargate.sgv2.jsonapi.service.shredding.model.DocumentId;
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.function.Supplier;
 
 public record UpdateOperationPage(
-    List<ReadAndUpdateOperation.UpdatedDocument> updatedDocuments, boolean returnDocs)
+    int matchedCount,
+    int modifiedCount,
+    List<ReadAndUpdateOperation.UpdatedDocument> updatedDocuments,
+    boolean returnDocs,
+    boolean moreDataFlag)
     implements Supplier<CommandResult> {
   @Override
   public CommandResult get() {
-    List<DocumentId> updatedIds = new ArrayList<>(updatedDocuments().size());
+    final DocumentId[] upsertedId = new DocumentId[1];
     List<JsonNode> updatedDocs = new ArrayList<>(updatedDocuments().size());
     updatedDocuments.forEach(
         update -> {
-          updatedIds.add(update.id());
-          updatedDocs.add(update.document());
+          if (update.upserted()) upsertedId[0] = update.id();
+          if (returnDocs) updatedDocs.add(update.document());
         });
+    Map<CommandStatus, Object> updateStatus = new HashMap<>();
```

Comment:
Could probably be `EnumMap` as key is enum?

Reply:
Done